### ISSUE TITLE
Autocomplete not working with arguments from static functions preloaded from another script

### ISSUE THREAD

**Issue by JoanPotatoes2021** [godotengine/godot#83466]
### Godot version

v4.1.2.stable.official [399c9dc39]

### System information

Windows 10

### Issue description

Autocomplete does not work when you press the spacebar for arguments from a static function preloaded from another script, requiring you to switch to that script to see the argument types, 

If you type a function from a object, the function autocompletes, if you press spacebar inside of that function, the expected arguments will pop up, but that do not happens if you try with a static function from a preloaded script.

![AutocompleteArgumentErrorGIF](https://github.com/godotengine/godot/assets/88800565/972bff7e-f03f-41bd-897b-e62897a07f9a)

This would be a great QoL improvement but also can be considered a bug? Because the default behavior is to display the arguments, just like the default functions, also everything is statically typed, unless I am missing something obvious the function should display the required arguments and their types.

Sorry if this is already an issue somewhere, from my search didn't found it or a solution, any workarounds are also welcomed if you know any.

### Steps to reproduce

See the MRP in the script editor to see the problem,

1 - Have a script with a static function saved as a script,
```
script file "cooking_things.gd"

extends Object
static func cook_something( ingredients:Array, recipe:Dictionary ) -> void: pass
    # Cook recipe.

```

2 - Preload that to a node to be used later:
`const COOK_SCRIPT = preload("cooking_things.gd")`

3 - Try to use that function later, you would expect to see what are the expected arguments.
`COOK_SCRIPT.cook_something( * )  # press space bar here will not display the arguments.`

### Minimal reproduction project

[autocomplete_arguments_error.zip](https://github.com/godotengine/godot/files/12922157/autocomplete_arguments_error.zip)

**Comment by HolonProduction** [comment]
Seems to be fixed in `master`. Can't pinpoint the exact commit though.

**Comment by akien-mga** [comment]
Closing as fixed in 4.2.
